src/routes/transactions/mappers/common/transaction-data.mapper.ts

```diff
@@ -108,61 +107,83 @@ export class TransactionDataMapper {
     dataDecoded: DataDecoded | null,
   ): Promise<Record<string, AddressInfo>> {
     if (dataDecoded === null || !isArray(dataDecoded.parameters)) return {};
+    const { method, parameters } = dataDecoded;
+    const promises: Promise<(AddressInfo | null)[] | AddressInfo | null>[] = [];
 
-    const addressInfos = await Promise.all(
-      this.getAddressParametersFromDataDecoded(dataDecoded).map((param) =>
-        this.getIfValidAddress(chainId, param),
-      ),
-    );
-
-    return addressInfos.reduce(
-      (addressInfoIndex, addressInfo) =>
-        addressInfo
-          ? { ...addressInfoIndex, [addressInfo.value]: addressInfo }
-          : addressInfoIndex,
-      {},
-    );
-  }
+    if (method === MULTI_SEND_METHOD_NAME) {
+      for (const parameter of parameters) {
+        const { name, valueDecoded } = parameter;
+        if (name === TRANSACTIONS_PARAMETER_NAME && valueDecoded) {
+          promises.push(this._getFromValueDecoded(chainId, valueDecoded));
+        }
+      }
+    } else {
+      for (const parameter of parameters) {
+        if (parameter.type === ADDRESS_PARAMETER_TYPE) {
+          promises.push(this._getIfValid(chainId, parameter.value));
+        }
+      }
+    }
```

Comment:
nitpick: I think it's more concise to have the above as follows:

```typescript
for (const parameter of parameters) {
  if (
    method === MULTI_SEND_METHOD_NAME &&
    parameter.name === TRANSACTIONS_PARAMETER_NAME &&
    parameter.valueDecoded
  ) {
    promises.push(
      this._getFromValueDecoded(chainId, parameter.valueDecoded),
    );
  } else if (parameter.type === ADDRESS_PARAMETER_TYPE) {
    promises.push(this._getIfValid(chainId, parameter.value));
  }
}
```

Reply:
Great, I agree it's simpler that way. Thanks, I've committed your suggestion 🚀